From adcb52796f4886d2831c8b21e27ba26ccb712094 Mon Sep 17 00:00:00 2001
From: Chaitanya Vadrevu <chaitanya.vadrevu@ni.com>
Date: Fri, 11 Aug 2023 15:24:56 -0500
Subject: [PATCH] exported: authorize all methods by default

Without polkit, no method is authorized by default.
We will be using dbus rules to control users' access to individual
methods. So authorize all methods by default in bolt.

Upstream-Status: Inappropriate [NI-specific changes]

Signed-off-by: Chaitanya Vadrevu <chaitanya.vadrevu@ni.com>
---
 boltd/bolt-exported.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/boltd/bolt-exported.c b/boltd/bolt-exported.c
index fdbb5ca..c491e91 100644
--- a/boltd/bolt-exported.c
+++ b/boltd/bolt-exported.c
@@ -593,14 +593,7 @@ handle_authorize_method_default (BoltExported          *exported,
                                  GDBusMethodInvocation *inv,
                                  GError               **error)
 {
-  const char *method_name;
-
-  method_name = g_dbus_method_invocation_get_method_name (inv);
-  g_set_error (error, G_DBUS_ERROR, G_DBUS_ERROR_ACCESS_DENIED,
-               "bolt operation '%s' denied by default policy",
-               method_name);
-
-  return FALSE;
+  return TRUE;
 }
 
 static gboolean
